mip/mip.go

```diff
@@ -0,0 +1,26 @@
+package mip
+
+// NewSolveOptions returns default solver options.
+func NewSolveOptions() SolveOptions {
+	connect()
+	return newSolveOptions()
+}
+
+// NewModel creates an empty MIP model.
+func NewModel() Model {
+	connect()
+	return newModel()
+}
+
+// NewSolver returns a new Solver which will use a solver
+// implemented by provider.
```

Comment:
```suggestion
// NewSolver returns a new Solver implemented by the given provider.
```
Maybe reduce repetition of solver here?